group-coordinator/src/main/java/org/apache/kafka/coordinator/group/consumer/CurrentAssignmentBuilder.java

```diff
@@ -251,195 +255,51 @@ private ConsumerGroupMember transitionToNewTargetAssignmentState() {
         }
 
         if (!newPartitionsPendingRevocation.isEmpty()) {
-            // If the partition pending revocation set is not empty, we transition the
-            // member to revoking and keep the current epoch. The transition to the new
-            // state is done when the member is updated.
+            // If there are partitions to be revoked, the member remains in its current
+            // epoch and requests the revocation of those partitions. It transitions to
+            // the UNREVOKED_PARTITIONS state to wait until the client acknowledges the
```

Comment:
"acknowledges the revocation of the partitions" -> this acknowledgement is done by removing these partitions from `ownedPartitions` right?

Reply:
Correct.